### DISCUSSION TITLE
Should pipes abort early by default?

### DISCUSSION THREAD

**Discussion by IlyaSemenov** [open-circle/valibot#624] [Polls]
Following #618, I am wondering why pipes don't abort early by default? What I mean is, take the example from [valibot documentation](https://valibot.dev/guides/parse-data/#configuration):

```ts
  const EmailSchema = v.pipe(v.string(), v.email(), v.endsWith('@example.com'));
  const email = v.parse(EmailSchema, 'jane@example.com', {
    abortPipeEarly: true,
  });
```

Naturally, there is no point in testing (and reporting) the string ending if it's not even an email address, and that's an inner property of a _particular pipe_, not the whole schema. Arguably, the majority of pipes will be structured similarly, and only select few pipes will test different aspects independently. For those cases, I'd say pipe is not even a good name. In real life, if something is stuck in the pipe (=fails validation) it's not moved further along. If liquid goes in parallel, it's a [manifold](https://en.wikipedia.org/wiki/Manifold_(fluid_mechanics)), not a pipe.

I propose to set `abortPipeEarly` by default, and allow to revert this with `v.config()` — opposite to how it works now.

**Comment by fabian-hiller** [comment]
For context, in the beginning Valibot only returned a single issue for a path, and later expanded that because we had several people requesting it. I think there are legitimate examples for both preferences. For anyone who prefers `abortPipeEarly` to be `true` by default, you can add the following line to your project to change the default configuration.

```ts
import * as v from 'valibot';

v.setGlobalConfig({ abortPipeEarly: true });
```

**Reply by IlyaSemenov** [reply]
Thanks for the heads up. It didn't occur to me that `setGlobalConfig` could change this behaviour, too. 

With that in mind, we shouldn't change the default, or it'll be kind of a footgun. If `abortPipeEarly: true` were the default and everyone was relying on it (like in `EmailSchema` example), then changing it in global config would make many schemas emit weird/duplicate errors or even crash.

So it should be either kept as is, or it should be early aborting by default without having a way to override this in global config.
